Homeworks/Homework_2/task.py

```diff
@@ -0,0 +1,131 @@
+import requests as re
+import csv
+import copy
+from collections import Counter
+import time as t
+from datetime import timedelta, date
+
+
+class CinemaUser:
+    HEADERS = {
+        "accept": "application/json",
+        "Authorization": "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIzMTI3NGFmYTRlNTUyMjRjYzRlN2Q0NmNlMTNkOTZjOSIsInN1YiI6IjVkNmZhMWZmNzdjMDFmMDAxMDU5NzQ4OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.lbpgyXlOXwrbY0mUmP-zQpNAMCw_h-oaudAJB6Cn5c8"
+    }
+
+    def __init__(self, number):
+        self.data = []
+        self.collection_of_structures = []
+        self.fetch_data_from_desired_pages(number)
+
+    def get_response(self, number):
+        url = f'https://api.themoviedb.org/3/discover/movie?include_adult=false&include_video=false&sort_by=popularity.desc&page={number}'
+        return re.get(url, headers=self.HEADERS)
+
+    def fetch_data_from_desired_pages(self, end):
+        for i in range(1, end + 1):
+            response = self.get_response(i)
+            self.data.extend(response.json()['results'])
+
+    def get_all_data_from_page(self):
+        return self.data
+
+    def get_data_about_movies(self, start, end, step):
+        return self.data[start:end:step]
+
+    def get_the_most_popular_movie(self):
+        return max(self.data, key=lambda x: x['popularity'])['original_title']
+
+    def find_movie(self, find_text):
+        return [film['original_title'] for film in self.data if film['overview'].find(find_text) > 0]
+
+    def print_genres(self):
+        return set(genre for film in self.data for genre in film['genre_ids'])
+
+    def delete_movies_with_genre(self, deleted_genre):
+        return [film['original_title'] for film in self.data if deleted_genre not in film['genre_ids']]
+
+    def get_most_popular_genre(self):
+        return Counter([genre for film in self.data for genre in film['genre_ids']]).most_common(1)[0][0]
+
+    def group_movies(self):
+        common_movies = [{film['original_title'], movie['original_title']} for i, film in enumerate(self.data) for genre in film['genre_ids'] for movie in self.data[i+1::] if genre in movie['genre_ids']]
```

Comment:
**(the groups should not allow inserts)**

Reply:
self.data[i+1:] -- you don't need second colon.

Reply:
if you have too long comprehensions, break them to several lines by logical blocks:
common_movies = [(film['original_title'], movie['original_title']) 
                 for i, film in enumerate(self.data) 
                 for genre in film['genre_ids'] 
                 for movie in self.data[i+1:] 
                 if genre in movie['genre_ids']]

Reply:
try to play with set().intersection() here